src/main/java/net/william278/velocitab/packet/ScoreboardManager.java

```diff
@@ -85,27 +90,57 @@ private void dispatchPacket(@NotNull UpdateTeamsPacket packet, @NotNull Player p
-            } else {
-                plugin.log("Failed to get ProtocolizePlayer for player " + player.getUsername() + " (UUID: " + player.getUniqueId() + ")");
-            }
+            final ConnectedPlayer connectedPlayer = (ConnectedPlayer) player;
+            connectedPlayer.getConnection().write(packet);
         } catch (Exception e) {
             plugin.log("Failed to dispatch packet (is the client or server modded or using an illegal version?)", e);
         }
     }
 
+    private static final MethodHandle STATE_REGISTRY$clientBound;
+    private static final MethodHandle PACKET_REGISTRY$register;
+    private static final MethodHandle PACKET_MAPPING$map;
+
+    static {
+        final MethodHandles.Lookup lookup = MethodHandles.lookup();
+        try {
+            final MethodHandles.Lookup stateRegistryLookup = MethodHandles.privateLookupIn(StateRegistry.class, lookup);
+            STATE_REGISTRY$clientBound = stateRegistryLookup.findGetter(StateRegistry.class, "clientbound", StateRegistry.PacketRegistry.class);
+
+            final MethodType mapType = MethodType.methodType(StateRegistry.PacketMapping.class, Integer.TYPE, ProtocolVersion.class, Boolean.TYPE);
+            PACKET_MAPPING$map = stateRegistryLookup.findStatic(StateRegistry.class, "map", mapType);
+
+            final MethodHandles.Lookup packetRegistryLookup = MethodHandles.privateLookupIn(StateRegistry.PacketRegistry.class, lookup);
+            final MethodType registerType = MethodType.methodType(void.class, Class.class, Supplier.class, StateRegistry.PacketMapping[].class);
+            PACKET_REGISTRY$register = packetRegistryLookup.findVirtual(StateRegistry.PacketRegistry.class, "register", registerType);
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     public void registerPacket() {
+        StateRegistry registry = StateRegistry.PLAY;
+
         try {
-            Protocolize.protocolRegistration().registerPacket(
-                    UpdateTeamsPacket.MAPPINGS,
-                    Protocol.PLAY,
-                    PacketDirection.CLIENTBOUND,
-                    UpdateTeamsPacket.class
+            final StateRegistry.PacketMapping[] MAPPINGS = {
+                    (StateRegistry.PacketMapping) PACKET_MAPPING$map.invoke(0x47, MINECRAFT_1_13, false),
+                    (StateRegistry.PacketMapping) PACKET_MAPPING$map.invoke(0x4B, MINECRAFT_1_14, false),
+                    (StateRegistry.PacketMapping) PACKET_MAPPING$map.invoke(0x4C, MINECRAFT_1_15, false),
+                    (StateRegistry.PacketMapping) PACKET_MAPPING$map.invoke(0x55, MINECRAFT_1_17, false),
+                    (StateRegistry.PacketMapping) PACKET_MAPPING$map.invoke(0x58, MINECRAFT_1_19_1, false),
+                    (StateRegistry.PacketMapping) PACKET_MAPPING$map.invoke(0x56, MINECRAFT_1_19_3, false),
+                    (StateRegistry.PacketMapping) PACKET_MAPPING$map.invoke(0x5A, MINECRAFT_1_19_4, false)
```

Comment:
I think it'd make sense to have a static Map<> of packet ID bytes to minecraft version constants in the UpdateTeamsPacket class rather than here, just so all the logic for that packet itself is contained in that class, and only the registration/management logic is in here. The entries of that map could then be invoked and fed into the registration command below.